### ISSUE TITLE
[Request future] smart-card, tokens and other

### ISSUE THREAD

**Issue by SergeyDjam** [binarymaster/rdpwrap#91]
Добрый день. Есть ли возможность сделать возможным сделать возможным использования смарт-карт и токенов подключенных к удаленному серверу в сессии RDP?

**Comment by SergeyDjam** [comment]
Hello, good day! How can smart-card and tockens in RDP session connected from remote server?

**Comment by 3da** [comment]
А при чём тут RDP wrapper? Оригинальный RDP позволяет это осуществлять?

**Comment by SergeyDjam** [comment]
В том и проблема что оригинальный именно виндовый не может... В тоже время VirtualBox если в нем поднять Windows и расшарить рабочий стол средствами VB, то такой проблемы не возникает. 
Потому и вопрос на будущее, есть ли возможность "исправить" проблему оригинального RDP?

**Comment by 3da** [comment]
Ну значит эта задача будет закрыта по причине того что это не относится к проекту и противоречит цели его существования.

**Comment by SergeyDjam** [comment]
хм, а как этот  вопрос противоречит цели проекта?

**Comment by 3da** [comment]
> хм, а как этот вопрос противоречит цели проекта?

Я думаю что этот проект сделан для того чтобы дать людям у которых Home версии Windows возможность RDP доступа к компьютеру. Потому что такая возможность отсутствует в Home версиях из коробки.

**Comment by SergeyDjam** [comment]
> "В тоже время VirtualBox если в нем поднять Windows и расшарить рабочий стол средствами VB, то такой проблемы не возникает. "

Т.е. не проблема протокола?

**Comment by 3da** [comment]
> Т.е. не проблема протокола?

А где связь между Рабочим столом внутри VirtualBox и RDP?

**Comment by SergeyDjam** [comment]
Потому как VB также использует для доступа к ВМ протокол RDP

**Comment by 3da** [comment]
> Потому как VB также использует для доступа к ВМ протокол RDP

Я полон сомнений что это так. Возможно там какой-то расширенный RDP?

**Comment by SergeyDjam** [comment]
разве такой существует?

**Comment by 3da** [comment]
Гуглится VRDP. Так что наверное есть

**Comment by SergeyDjam** [comment]
да вы правы, видимо это только для VB актуально. Жаль

**Comment by 3da** [comment]
> да вы правы, видимо это только для VB актуально. Жаль

Существует же куча альтернативных утилит для удалённого доступа к рабочему столу? Может там есть что-то что будет удовлетворять вашим потребностям?

**Comment by SergeyDjam** [comment]
что-то на 8-ке толком ничего из опробаванного толком не работает.

**Comment by binarymaster** [comment]
Windows RDP по идее поддерживает смарт карты из коробки. Полагаю, главное чтобы перенаправление USB устройств работало.

А вот в каких сборках оно работает, а в каких нет - это совсем другой вопрос.

**Comment by SergeyDjam** [comment]
да, но только через проброс с клиента. Но если ключики стоят на сервере терминалов то они не видны. Такая вот особенность.

**Comment by binarymaster** [comment]
Да, это особенность Windows, и RDP Wrapper тут ничем помочь не сможет.

**Comment by SergeyDjam** [comment]
подумалось, если rdpwrapper позволяет сделать доступ к RDP там где его по умолчанию нет, то и остальное под силу...

**Comment by binarymaster** [comment]
Скажем так, внутри TS есть скрытые переключатели, которыми он и манипулирует.

Но есть такие вещи, которые вшиты намертво, которые нельзя переключить.

**Comment by SergeyDjam** [comment]
к сожалению, профан в этом вопросе... Спасибо за разъяснения

**Comment by R030t1** [comment]
You may wish to look at [this website](https://www.howtogeek.com/97987/how-to-forward-local-drives-to-remote-machines-using-remote-desktop/). If certain device classes are not supported then you will be unable to use them.

What VirtualBox does is make it look like, to the guest machine, that there is new hardware available. Doing what you wish to do requires a driver on the computer you are remoting in to to provide the hardware. [This is implemented and working under Linux, and perhaps Windows, for USB devices, however if the device is not USB you will need to search for something else.](https://www.spice-space.org/page/UsbRedir)
